tests/main.test.ts

```diff
@@ -169,19 +74,23 @@ describe("Action tests", () => {
         },
       },
     }));
+    const mergePullRequest = jest.fn();
+    jest.mock(githubHelpersPath, () => {
+      const actualModule = jest.requireActual(githubHelpersPath) as object;
+      return {
+        __esModule: true,
+        ...actualModule,
+        mergePullRequest,
+      };
+    });
     const run = (await import("../src/action")).run;
     await expect(run()).resolves.toMatchObject({ status: 200 });
-    const pullRequests = await dataSource.getRepository(PullRequest).find();
-    expect(pullRequests).toHaveLength(1);
+    expect(mergePullRequest).not.toHaveBeenCalled();
   });
 
   it("Should close a PR that is past the threshold", async () => {
-    const dataSource = await initializeDataSource(dbName);
-    const pr = new PullRequest();
-    pr.url = htmlUrl;
     const lastActivityDate = new Date();
     lastActivityDate.setDate(new Date().getDate() - 8);
-    await pr.save();
     server.use(
       http.get(
         "https://api.github.com/repos/:org/:repo/pulls/:id/merge",
```

Comment:
why do you mock 404 for merge, won't this mean the run will fail?
also why is there no mock for timeline events? it should return an event older than the threshold

Reply:
404 means that the pull request is not merged:
https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#check-if-a-pull-request-has-been-merged--status-codes

But you are correct, I didn't pay attention after I removed the db and the test was still passing. The timeline should indeed be mocked beyong the threshold, will fix.